adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -119,6 +118,480 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   def transform(tFn: RDD[T] => RDD[T]): U = {
     replaceRdd(tFn(rdd))
   }
+  val self = this
```

Comment:
self is required so that the SortedTrait can access the U object that contains it. In joins, we need to be able to leave the left RDD unchanged if it is already properly partitioned to avoid another shuffle phase. Is there another way to access <this> without a self reference in the outer trait?